apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx

```diff
@@ -215,14 +226,71 @@ export const CreateHookSheet = ({
       {
         onSuccess: () => {
           toast.success(`Successfully created ${values.hookType}.`)
+          if (statements.length > 0) {
```

Comment:
I think this logic is wrong or duplicated with the one at line 258. What is the idea, revoke permissions before changing, change then grant the permissions again, right?

Reply:
we just want to apply the permission changes to the postgres functions (old and new) when the config is updated

the old function will be restored to it's original state and the new function will have the new permissions applied

Reply:
the order in which it's being executed here doesn't really matter, as long as the permission changes are applied when the auth config update endpoint is called it's fine